controllers/complianceeventsapi/server.go

```diff
@@ -309,6 +329,30 @@ func (s *ComplianceAPIServer) Start(ctx context.Context, serverContext *Complian
 	}
 }
 
+func getUserKubeConfig(config *rest.Config, w http.ResponseWriter, r *http.Request) (*rest.Config, error) {
```

Comment:
This duplicates `getClientFromToken` a bit. Could you delete `getClientFromToken`  and refactor `canRecordComplianceEvent` to use this new function?